internal/pkg/composable/controller.go

```diff
@@ -192,8 +171,13 @@ func (c *controller) Run(ctx context.Context) error {
 		for {
 			select {
 			case <-ctx.Done():
-				cleanupFn()
 				return ctx.Err()
+			case observed := <-c.observedCh:
+				c.handleObserved(localCtx, &wg, stateChangedChan, fetchContextProviders, observed)
+				t.Reset(100 * time.Millisecond)
+				c.logger.Debugf("Observed state changed for composable inputs; debounce started")
+				drainChan(stateChangedChan)
+				break DEBOUNCE
```

Comment:
This is so we always update vars after `Observe` is called? If so, we should make sure we only emit if anything changed.

Reply:
This should only happen when the policy changes which should not be often, at least not as often compared to a variable changing in a dynamic provider like kubernetes. I will update it to only emit if something actually changed.

Reply:
Updated in latest commit.